src/PHPCR/Util/UUIDHelper.php

```diff
@@ -29,31 +32,12 @@ public static function isUUID($id)
     /**
      * Generate a UUID.
      *
-     * This UUID can not be guaranteed to be unique within the repository.
-     * Ensuring this is the responsibility of the repository implementation.
-     *
      * @return string a random UUID
      */
     public static function generateUUID()
     {
-        return sprintf('%04x%04x-%04x-%04x-%04x-%04x%04x%04x',
-            // 32 bits for "time_low"
-            mt_rand(0, 0xffff), mt_rand(0, 0xffff),
-
-            // 16 bits for "time_mid"
-            mt_rand(0, 0xffff),
-
-            // 16 bits for "time_hi_and_version",
-            // four most significant bits holds version number 4
-            mt_rand(0, 0x0fff) | 0x4000,
-
-            // 16 bits, 8 bits for "clk_seq_hi_res",
-            // 8 bits for "clk_seq_low",
-            // two most significant bits holds zero and one for variant DCE1.1
-            mt_rand(0, 0x3fff) | 0x8000,
+        $uuid4 = Uuid::uuid4();
```

Comment:
please keep the old code and simply check dynamically if the ``Rhumsaa\Uuid\Uuid`` class is available